src/integrationtest/java/com/ericsson/gerrit/plugins/eiffel/integrationtest/ServiceIntegrationSteps.java

```diff
@@ -0,0 +1,260 @@
+				+ "\"blockedContentType\":{\"values\":[]},\"blockedContentTypeWhitelist\":"
+				+ "{\"value\":\"false\"},\"blockedFileExtension\":{\"values\":[]},"
+				+ "\"blockedKeywordPattern\":{\"values\":[]},\"invalidFilenamePattern\":"
+				+ "{\"values\":[]},\"maxPathLength\":{\"value\":\"0\"},\"project\":"
+				+ "{\"values\":[]},\"ref\":{\"values\":[]},\"rejectDuplicatePathnames\":"
+				+ "{\"value\":\"false\"},\"rejectDuplicatePathnamesLocale\":{\"value\":\"en\"},"
+				+ "\"rejectSubmodule\":{\"value\":\"false\"},\"rejectSymlink\":"
+				+ "{\"value\":\"false\"},\"rejectWindowsLineEndings\":{\"value\":\"false\"},"
+				+ "\"requiredFooter\":{\"values\":[]}}}}";
+
+		HttpRequest request = new HttpRequest(HttpMethod.PUT);
+		ResponseEntity response;
+		response = request.setHeader("Cookie", gerritAccountCookie)
+				.setHeader("Content-Type", "application/json; charset=UTF-8").setHeader("Accept", "application/json")
+				.setHeader("X-Gerrit-Auth", gerritXAuthToken).setBaseUrl(GERRIT_BASE_URL)
+				.setEndpoint("/projects/" + projectName + "/config").setBody(jsonBodyString).performRequest();
+
+		return response;
+	}
+
+	private ResponseEntity submitChange() throws URISyntaxException, ClientProtocolException, IOException {
+		String jsonBodyString = "{\"project\":\"" + projectName + "\" ,"
+				+ "\"subject\" : \"Let's support 100% Gerrit workflow direct in browser\", "
+				+ "\"topic\" : \"create-change-in-browser\", " + "\"status\" : \"DRAFT\", " + "\"branch\":\"master\"}";
+
+		HttpRequest request = new HttpRequest(HttpMethod.POST);
+		ResponseEntity response;
+		response = request.setHeader("Cookie", gerritAccountCookie)
+				.setHeader("Content-Type", "application/json; charset=UTF-8").setHeader("Accept", "application/json")
+				.setHeader("X-Gerrit-Auth", gerritXAuthToken).setBaseUrl(GERRIT_BASE_URL).setEndpoint("/changes/")
+				.setBody(jsonBodyString).performRequest();
+
+		return response;
+	}
+
+	protected List<String> consumeMessages(int messageCount, long timeout) throws IOException, TimeoutException {
+		List<String> messages = new ArrayList<String>();
+		ConnectionFactory factory = new ConnectionFactory();
+		factory.setHost(RABBITMQ_HOSTNAME);
+		factory.setPort(RABBITMQ_PORT);
+		factory.setUsername(RABBITMQ_USERNAME);
+		factory.setPassword(RABBITMQ_PASSWORD);
+		Connection connection = factory.newConnection();
+		Channel channel = connection.createChannel();
+
+		long stopTime = System.currentTimeMillis() + timeout;
+
+		while (System.currentTimeMillis() < stopTime) {
+			try {
+			    LOGGER.error("RabbitMQ queue is:" + RABBITMQ_QUEUENAME);
+				GetResponse response = channel.basicGet(RABBITMQ_QUEUENAME, true);
+				if (response != null) {
+					messages.add(new String(response.getBody(), "UTF-8"));
+				}
+
+				if (messages.size() == messageCount) {
+					return messages;
+				}
+			} catch (Exception e) {
+				LOGGER.error("RabbitMQ failed to get from queue", e);
```

Comment:
Do we need to cleanup ConnetionFactory and Connections objects here?

Reply:
Done in upcoming change

Reply:
ok